src/Microsoft.ML.TimeSeries/IidAnomalyDetectionBase.cs

```diff
@@ -13,99 +13,147 @@
 namespace Microsoft.ML.TimeSeriesProcessing
 {
     /// <summary>
-    /// This transform computes the p-values and martingale scores for a supposedly i.i.d input sequence of floats. In other words, it assumes
+    /// The wrapper to transform that computes the p-values and martingale scores for a supposedly i.i.d input sequence of floats. In other words, it assumes
     /// the input sequence represents the raw anomaly score which might have been computed via another process.
     /// </summary>
-    public abstract class IidAnomalyDetectionBase : SequentialAnomalyDetectionTransformBase<Single, IidAnomalyDetectionBase.State>
+    public class IidAnomalyDetectionBaseWrapper : IStatefulTransformer, ICanSaveModel
     {
-        public IidAnomalyDetectionBase(ArgumentsBase args, string name, IHostEnvironment env)
-            : base(args, name, env)
-        {
-            InitialWindowSize = 0;
-            StateRef = new State();
-            StateRef.InitState(WindowSize, InitialWindowSize, this, Host);
-        }
+        public bool IsRowToRowMapper => Base.IsRowToRowMapper;
 
-        public IidAnomalyDetectionBase(IHostEnvironment env, ModelLoadContext ctx, string name)
-            : base(env, ctx, name)
-        {
-            Host.CheckDecode(InitialWindowSize == 0);
-            StateRef = new State(ctx.Reader);
-            StateRef.InitState(this, Host);
-        }
+        IStatefulTransformer IStatefulTransformer.Clone() => Base.Clone();
+
+        public Schema GetOutputSchema(Schema inputSchema) => Base.GetOutputSchema(inputSchema);
```

Comment:
`Base` doesn't sound a good name because it implies `IidAnomalyDetectionBaseWrapper` is a kind of `Base`. Could we do `InternalTransformer` or something like that?